[DOCS-5328] Add beta docs for Oracle (#17828)
DataDog/documentation: content/en/database_monitoring/setup_oracle/selfhosted.md

---

> title: Setting Up Database Monitoring for self-hosted Oracle

Reviewer: ```suggestion
title: Setting Up Database Monitoring for Self-Hosted Oracle
```

kind: documentation
description: Install and configure Database Monitoring for self-hosted SQL Server
further_reading:
- link: "/integrations/oracle/"
  tag: "Documentation"
  text: "Basic Oracle Integration"

---

{{< site-region region="gov" >}}
<div class="alert alert-warning">Database Monitoring is not supported for this site.</div>
{{< /site-region >}}

<div class="alert alert-info">

> The features described on this page are in beta. Contact your Customer Success Manager to get the required agent build and installation directions.

Reviewer 2: capitalize Agent

> The features described on this page are in beta. Contact your Customer Success Manager to get the required agent build and installation directions.

Reviewer: ```suggestion
The features described on this page are in private beta. Contact your Customer Success Manager for information on the required Agent build and installation instructions.
```

</div>

Database Monitoring provides deep visibility into your Oracle databases by exposing query samples to profile your different workloads and diagnose issues quickly.

> Do the following steps to enable Database Monitoring with your database:

Reviewer: ```suggestion
Complete the following steps to enable Database Monitoring with your database:
```

> 1. [Grant the Agent access to the database](#grant-the-agent-access)
> 2. [Install the Agent](#install-the-agent)
> 

Reviewer: ```suggestion
1. [Grant the Agent access to the database](#grant-the-agent-access).
2. [Install the Agent](#install-the-agent).

```

## Before you begin

Supported Oracle versions
: 19c, 21c

Supported Agent versions

> : Please contact your Customer Success Manager to get agent installation directions for the beta.

Reviewer 2: capitalize Agent

> : Please contact your Customer Success Manager to get agent installation directions for the beta.

Reviewer: ```suggestion
For more information on the required Agent build, contact your Customer Success Manager.
```

## Grant the Agent access

The Datadog Agent requires read-only access to the database server in order to collect samples.

Create a read-only login to connect to your server and grant the required permissions:

```SQL
CREATE USER c##datadog IDENTIFIED BY &password CONTAINER = ALL ;

ALTER USER c##datadog SET CONTAINER_DATA=ALL CONTAINER=CURRENT;

grant create session to c##datadog ;
grant select on v_$session to c##datadog ;
grant select on v_$database to c##datadog ;
grant select on v_$containers to c##datadog;
grant select on v_$sqlstats to c##datadog ;
grant select on v_$instance to c##datadog ;
grant select on dba_feature_usage_statistics to c##datadog ;
```

> Create a new `view`, and give the agent user access to it:

Reviewer 2: capitalize Agent

        4294967291, TO_NUMBER(NULL), BITAND(s.ksusefblocker, 65535)
    ) as final_blocking_session,
    DECODE(w.kslwtinwait,
        1, 'WAITING', decode(bitand(w.kslwtflags, 256), 0, 'WAITED UNKNOWN TIME',
        decode(round(w.kslwtstime / 10000), 0, 'WAITED SHORT TIME', 'WAITED KNOWN TIME'))
    ) as STATE,
    e.kslednam as event,
    e.ksledclass as wait_class,
    w.kslwtstime as wait_time_micro,
    c.name as pdb_name,
    sq.sql_text as sql_text,
    sq.sql_fulltext as sql_fulltext,
    sq_prev.sql_fulltext as prev_sql_fulltext,
    comm.command_name
FROM
  x$ksuse s,
  x$kslwt w,
  x$ksled e,
  v$sqlstats sq,
  v$sqlstats sq_prev,
  v$containers c,
  v$sqlcommand comm
WHERE
  BITAND(s.ksspaflg, 1) != 0
  AND BITAND(s.ksuseflg, 1) != 0
  AND s.inst_id = USERENV('Instance')
  AND s.indx = w.kslwtsid
  AND w.kslwtevt = e.indx
  AND s.ksusesqi = sq.sql_id(+)
  AND s.ksusesph = sq.plan_hash_value(+)
  AND s.ksusepsi = sq_prev.sql_id(+)
  AND s.ksusepha = sq_prev.plan_hash_value(+)
  AND s.con_id = c.con_id(+)
  AND s.ksuudoct = comm.command_type(+)
;

GRANT SELECT ON dd_session TO c##datadog ;
```

> Grant additional permissions if you wish to use the basic oracle integration as well:

Reviewer 2: capitalize Oracle

```SQL
GRANT CREATE SESSION TO c##datadog CONTAINER=ALL;
Grant select any dictionary to c##datadog container=all;
GRANT SELECT ON GV_$PROCESS TO c##datadog CONTAINER=ALL;
GRANT SELECT ON gv_$sysmetric TO c##datadog CONTAINER=ALL;
```

## Install the Agent

> To start collecting Oracle telemetry, first [install the Datadog Agent][1]. In order to install the correct version to participate in the beta, please contact your Customer Success Manager.

Reviewer: ```suggestion
To start collecting Oracle telemetry, first [install the Datadog Agent][1]. Note that you must install the correct version to participate in the beta. Contact your Customer Success Manager for more information.
```

Create the Oracle Agent conf file `/etc/datadog-agent/conf.d/oracle-dbm.d/conf.yaml`. See the [sample conf file][2] for all available configuration options.

```yaml
init_config:
instances:
  - dbm: true
    server: '<HOSTNAME>:<SQL_PORT>'
    service_name: "<SERVICE_NAME>" # The Oracle CDB service name
    username: 'c##datadog'
    password: '<PASSWORD>'
    tags:  # Optional
      - 'service:<CUSTOM_SERVICE>'
      - 'env:<CUSTOM_ENV>'
```

> Use the `service` and `env` tags to link your database telemetry to other telemetry through a common tagging scheme. See [Unified Service Tagging][3] on how these tags are used throughout Datadog.

Reviewer: ```suggestion
Use the `service` and `env` tags to link your database telemetry to other telemetry through a common tagging scheme. See [Unified Service Tagging][3] to learn more about how these tags are used in Datadog.
```

Once all Agent configuration is complete, [restart the Datadog Agent][4].

> To install the basic Oracle integration, you should create an additional Oracle Agent conf file at `/etc/datadog-agent/conf.d/oracle.d/conf.yaml`

Reviewer: ```suggestion
To install the basic Oracle integration, you should create an additional Oracle Agent conf file at `/etc/datadog-agent/conf.d/oracle.d/conf.yaml`.
```

Reviewer: I'm not too familiar with our Oracle integration(s). Do we have more than one? If not, we can probably remove the word "basic".

Author: yeah there is a regular oracle integration and this new one that we have written (eventually they will be merged), and they can be thought of as different feature sets
